app/packages/looker/src/worker/threed-label-processor.ts

```diff
@@ -139,35 +108,67 @@ const PainterFactory3D = (
    * Impute bounding box parameters.
    */
   Detection: (label: DetectionLabel) => {
-    const {
-      width: canvasWidth,
-      height: canvasHeight,
-      min_bound,
-      max_bound,
-    } = orthographicProjectionParams;
-    const [xmin, ymin] = min_bound;
-    const [xmax, ymax] = max_bound;
-
-    const [x, y, z] = label.location; // centroid of bounding box
-    const [lx, ly, lz] = label.dimensions; // length of bounding box in each dimension
-
-    const { xScale, yScale } = getScalingFactorForLabel(
-      label._id,
-      canvasWidth,
-      canvasHeight,
-      xmin,
-      xmax,
-      ymin,
-      ymax
-    );
+    const { min_bound, max_bound, normal } = orthographicProjectionParams;
+    const [xmin, ymin, zmin] = min_bound;
+    const [xmax, ymax, zmax] = max_bound;
+
+    const [lx, ly, lz] = label.location; // centroid of bounding box
+    const [dx, dy, dz] = label.dimensions; // length of bounding box in each dimension
+    const [rx, ry, rz] = label.rotation ?? [0, 0, 0]; // rotation of bounding box
+
+    const [nx, ny, nz] = normal;
+
+    const box: BoundingBox3D = {
+      dimensions: [dx, dy, dz],
+      location: [lx, ly, lz],
+      rotation: [rx, ry, rz],
+    };
+
+    let projectionPlane: "xy" | "xz" | "yz" = "xy";
```

Comment:
Me: after writing js for multiple decades I had a tear in my when I finally saw that typescript lets us define magic strings as enums removing meaningless random magic strings

sashank: